app/internal/page/share_code.go

```diff
@@ -0,0 +1,52 @@
+package page
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/notify"
+)
+
+type ShareCodeData struct {
+	SessionID string
+	LpaID     string
+}
+
+type shareCodeSender struct {
+	dataStore    DataStore
+	notifyClient NotifyClient
+	appPublicURL string
+	randomString func(int) string
+}
+
+func NewShareCodeSender(dataStore DataStore, notifyClient NotifyClient, appPublicURL string, randomString func(int) string) *shareCodeSender {
+	return &shareCodeSender{
+		dataStore:    dataStore,
+		notifyClient: notifyClient,
+		appPublicURL: appPublicURL,
+		randomString: randomString,
+	}
+}
+
+func (s *shareCodeSender) Send(ctx context.Context, template notify.TemplateId, appData AppData, email string) error {
```

Comment:
I hate having a Xer.X but it has cleaned up the code

Reply:
It's a small price to pay